submissions/iakow/OOP-exercise/index.js

```diff
@@ -0,0 +1,97 @@
+class Inhabitant {
+   constructor({ species, name, say, gender }) {
+      this.species = species;
+      this.name = name;
+      this.say = say;
+      this.gender = gender;
+      this.friends = new Set();
+   }
+
+   makeFriends(...applicants) {
+      applicants.filter(applicant => applicant instanceof Inhabitant && applicant != this)
+         .forEach(applicant => this.friends.add(applicant));
+   }
+
+   toString() {
+      const { name, species, friends, gender } = this;
+      return (
+         `Data: ${species}, ${gender}, ${name}, ${friends.size || 'no'} friend` +
+         `${friends.size == 1 ? '' : 's'}`
+      )
+   }
+}
+
+class Animal extends Inhabitant {
+   constructor({ species, name, gender, say, paws = 4 }) {
+      super({ species, name, say, gender });
+      this.paws = paws;
+      this.gender = gender;
```

Comment:
Why do we need re-initialize gender after it is being passed to the parent class constructor?
Check also other classes and fix.